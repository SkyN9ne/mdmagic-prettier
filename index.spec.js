const format = require('./index.js');

function wrapCode(code, header = '```js', footer = '```') {
  return [header, code, footer].join('\n');
}

describe('markdown-magic-prettier', () => {
  it('formats code blocks', () => {
    const code = 'console.log("hello world")'
    expect(format(wrapCode(code))).toMatchSnapshot();
  });

  it('returns original content if codeblock is not found', () => {
    expect(format('foo bar')).toMatchSnapshot()
    expect(format('```')).toMatchSnapshot()
  })

  it('adds a closing tag if it is forgotten', () => {
    const code = 'console.log("hello world")'
    expect(format(wrapCode(code, '```', ''))).toMatchSnapshot()
  });
})
