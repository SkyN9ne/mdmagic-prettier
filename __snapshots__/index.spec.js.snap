// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown-magic-prettier adds a closing tag if it is forgotten 1`] = `
"\`\`\`js
console.log('hello world');
\`\`\`"
`;

exports[`markdown-magic-prettier formats code blocks 1`] = `
"\`\`\`js
console.log('hello world');
\`\`\`"
`;

exports[`markdown-magic-prettier returns original content if codeblock is not found 1`] = `"foo bar"`;

exports[`markdown-magic-prettier returns original content if codeblock is not found 2`] = `"\`\`\`"`;
